### DISCUSSION TITLE
How to access content within an iframe?

### DISCUSSION THREAD

**Discussion by dinameda** [getgauge/taiko#2493] [General]
I have an iframe which has the structure as shown - 
`<iframe _ngcontent-jhv-c568="" frameborder="0" class="k-iframe">
     #document
     <html><body><div>content</div></body></html>
</iframe>`
To access the content within the iframe, we use the switchTo() method in chromedriver in Selenium to switch focus to the iframe from main browser. 
Is there an equivalent api in Taiko to read/update the iframe content?

**Comment by NivedhaSenthil** [comment]
We should be able to directly access elements within iFrames using apis and no extra config should be needed. Let us know if that doesn't work

**Reply by dinameda** [reply]
I tried directly accessing the iframe using evalute method, but not able to read the content within the iframe. I can get to the iframe, not the html within the iframe. I have pasted the code snippet and the iframe below. Thanks for your help.


`
const editorPath = `//kendo-editor//div//iframe`;
var editorFrame = await evaluate($(editorPath), (element) => {
    await switchTo(element);
    var content = await evaluate($(`//body//div//p`), (pElm) => {
         return pElm.innerHTML;
    });
    return content;
});
`

<div _ngcontent-aim-c0="" class="k-editor-content">
	<iframe _ngcontent-aim-c0="" class="k-iframe" frameborder="0" srcdoc="<!DOCTYPE html>">
		#document
			<html>
				<head>
					<meta charset="utf-8">
					<style></style>
				</head>
				<body>
					<div contenteditable="true" translate="no" tabindex="" class="ProseMirror">
						<p>The Kendo UI Angular Editor allows your users to edit HTML in a familiar, user-friendly way.<br>
							In this version, the Editor provides the core HTML editing engine which includes basic text formatting, hyperlinks, and lists.
						</p>
						<div style="display: inline-block; width: 39%;">
							<p>Features include:</p>
							<ul>
								<li><p>Text formatting</p></li>
								<li><p>Bulleted and numbered lists</p></li>
								<li><p>Hyperlinks</p></li>
								<li><p>Cross-browser support</p></li>
							</ul>
						</div>
					</div>
				</body>
			</html>
	</iframe>
</div>
